### PR TITLE
Authorization: Changes for role assignment changelog list

### PR DESCRIPTION
**Pull request by namratab**
This command gives the change history wrt Microsoft.Authorization provider by querying the Insights service. The results and post-processed to resolve object ids and make it presentable.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@namratab__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (nbaliga). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by namratab** [comment]
Tests unrelated to my change are failing - is there a workaround ?
https://travis-ci.org/Azure/azure-xplat-cli/jobs/79576852 

arm group template download should download template file using name of template:

1282     Uncaught Error: Nock: No match for HTTP request GET /Microsoft.Gallery/galleryitems/Microsoft.WebSiteMySQLDatabase.0.5.0

File: lib/commands/arm/role/role.assignment.audit._js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+var profile = require('../../../util/profile');
+var adUtils = require('../ad/adUtils');
+var rbacClients = require('../role/rbacClients');
+
+var $ = utils.getLocaleString;
+
+exports.init = function (cli) {
+  var log = cli.output;
+  var role = cli.category('role');
+  var roleAssignmentComand = role.category('assignment');
```
Comment (amarzavery) [inline lib/commands/arm/role/role.assignment.audit._js]: nit pick.. Command spelling

File: lib/commands/arm/role/role.assignment.audit._js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  var roleAssignmentLogsComand = roleAssignmentComand.category('changelog');
+
+  roleAssignmentLogsComand.command('list')
+      .description($('List access change history for the subscription for the specified time i.e. who granted(or revoked) what role to whom at what scope'))
+      .usage('[options]')
+      .option('-b --startTime <startTime>', $('The start time of the query. If both startTime and endTime are not specified, defaults to the last one hour. Maximum query interval supported is 15 days.'))
```
Comment (amarzavery) [inline lib/commands/arm/role/role.assignment.audit._js]: Can you provide an example of start time and end time
- YYYY/MM/DD:HH:MM:SS etc.

File: test/commands/arm/role/arm.role.assignment.audit-tests.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        });
+      });
+      
+      afterEach(function (done) {
+        suite.teardownTest(done);
+      });
+      
+      describe('list', function () {
+       it('should work', function (done) {
+          suite.execute('role assignment changelog list -b %s -e %s --json', '2015-09-08', '2015-09-09',function (result) {
```
Comment (amarzavery) [inline test/commands/arm/role/arm.role.assignment.audit-tests.js]: Will this test run live all the time, at the end of every release activities?

File: lib/commands/arm/role/role.assignment.audit._js (hunk 1 above)
Comment (namratab) [inline lib/commands/arm/role/role.assignment.audit._js]: Fixed, thanks

File: test/commands/arm/role/arm.role.assignment.audit-tests.js (hunk 3 above)
Comment (namratab) [inline test/commands/arm/role/arm.role.assignment.audit-tests.js]: Fixed the test per offline discussion

**Pull request merged**